packages/lambda-powertools-http-client/index.js

```diff
@@ -34,7 +34,10 @@ function getRequest (uri, method) {
 function setHeaders (request, headers) {
   const headerNames = Object.keys(headers)
   headerNames.forEach(h => {
-    request = request.set(h, headers[h])
+    const headerValue = headers[h]
+    if (headerValue) {
```

Comment:
This may be slightly safer in some instances... 
```suggestion
    if (!!headerValue) {
```